### PR TITLE
Use authentication with Docker Hub

### PR DESCRIPTION
**Pull request by xhlulu**
## Pre-Merge checklist
- [ ] The project was correctly built with `npm run build:all`.
- [ ] If there was any conflict, it was solved correctly.
- [ ] All changes were documented in CHANGELOG.md.
- [ ] All tests on CircleCI have passed.
- [ ] All Percy visual changes have been approved.
- [ ] Two people have :dancer:'d the pull request. You can be one of these people if you are a Dash Cytoscape core contributor.

## Reference Issues

Closes #[issue number]

## Other comments

### CODE REVIEW

**Review by rpkyle — CHANGES_REQUESTED** [review]
Looks good, now I think you just need to add the context in your workflow though, so after `jobs:` you would add

```
context: dash-docker-hub
```

**Review by rpkyle — APPROVED** [review]
@xhlulu Approved once tests pass and you move the `context:` line right below each of the job names, sorry, I should have been a bit clearer -- will look something like this:

image

File: .circleci/config.yml [hunk 1]
```diff
@@ -121,9 +121,9 @@ workflows:
     version: 2
     build:
         jobs:
-            - "python-3.6"
+            - "python-3.6":
```
Comment (rpkyle) [inline .circleci/config.yml:124]: :tada: :woman_dancing: once tests pass. Thanks!

**Review by rpkyle — APPROVED** [review]

**Comment by xhlulu** [comment]
```
======================================================================
ERROR: setUpClass (tests.test_callbacks.Tests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/circleci/project/tests/IntegrationTests.py", line 30, in setUpClass
    cls.driver = webdriver.Chrome(options=options)
  File "/home/circleci/project/venv/lib/python3.7/site-packages/selenium/webdriver/chrome/webdriver.py", line 81, in __init__
    desired_capabilities=desired_capabilities)
  File "/home/circleci/project/venv/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 157, in __init__
    self.start_session(capabilities, browser_profile)
  File "/home/circleci/project/venv/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 252, in start_session
    response = self.execute(Command.NEW_SESSION, parameters)
  File "/home/circleci/project/venv/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/home/circleci/project/venv/lib/python3.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: unknown error: Chrome failed to start: exited abnormally.
  (unknown error: DevToolsActivePort file doesn't exist)
  (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)
```

Seems like the tests are crashing now

**Comment by xhlulu** [comment]
Do you want to discuss that on Tuesday (since you are on :palm_tree: )

**Comment by rpkyle** [comment]
> Do you want to discuss that on Tuesday (since you are on palm_tree )

I'm away next week, but this failure appears unrelated to the Docker auth edits (since your image pulls appear to complete successfully). I'm curious if it's related to a previously reported ChromeDriver issue: https://github.com/SeleniumHQ/selenium/issues/6049.

**Comment by xhlulu** [comment]
The issue was reported in 2018-2019 and the tests seemed to work until august 2020.

**Comment by rpkyle** [comment]
> The issue was reported in 2018-2019 and the tests seemed to work until august 2020.

FWIW, I think we're OK to merge the Docker auth edits here, even if the integration test is failing. We can open an issue to address those separately, but if you'd prefer to wait, that's OK too. I don't think the Docker policy change goes into effect until November 1st, so there's still a little time.

**Pull request merged**